test/check-application

```diff
@@ -2259,12 +2279,61 @@ class TestFiles(testlib.MachineCase):
-            b.click(".pf-v5-c-alert__action button")
-            b.wait_not_present(".pf-v5-c-alert__action")
+            assert_upload_alert([os.path.basename(files[-1])], "admin:admin")
 
             self.assertEqual(m.execute(f"cat {dest_dir}/{filename}"), "new content")
 
+            # As administrator upload in testuser and change ownership to root:root
+
+            m.execute("useradd --create-home testuser")
+            b.go('/files#/?path=/home/testuser')
+            b.wait_not_present('.pf-v5-c-empty-state')
+
+            testuser_file = str(Path(tmpdir) / "testuser.txt")
+            test_content = "testdata"
+            with open(testuser_file, "w") as fp:
+                fp.write(test_content)
+
+            testuser_filename = os.path.basename(testuser_file)
+            b.click("#upload-file-btn")
+            b.upload_files("#upload-file-btn + input[type='file']", [testuser_file])
+            assert_upload_alert([testuser_filename], "testuser:testuser", close=False)
+            self.assert_owner(f'/home/testuser/{testuser_filename}', 'testuser:testuser')
+            contents = m.execute(f"cat /home/testuser/{testuser_filename}")
+            self.assertEqual(contents, test_content)
+
+            b.click(".pf-v5-c-alert__action-group button")
+            b.wait_not_present(".pf-v5-c-alert__action")
+            b.wait_in_text(".pf-v5-c-modal-box__title-text", testuser_filename)
+            b.wait_val("#edit-permissions-owner", "testuser")
+            b.wait_val("#edit-permissions-group", "testuser")
+            b.wait_val("#edit-permissions-owner-access", "read-write")
+            b.wait_val("#edit-permissions-group-access", "read-only")
+            b.wait_val("#edit-permissions-other-access", "read-only")
+
+            b.select_from_dropdown("#edit-permissions-owner", "root")
+            b.select_from_dropdown("#edit-permissions-group", "root")
+            b.click("button.pf-m-primary")
+            b.wait_not_present(".pf-v5-c-modal-box")
+
+            self.assert_owner(f'/home/testuser/{testuser_filename}', 'root:root')
+
+            # Uploading to an immutable dir fails to create a temp file, check that it is cleaned up
+
+            m.execute("chattr +i /home/testuser")
+            self.addCleanup(m.execute, "chattr -i /home/testuser")
+            b.click("#upload-file-btn")
+            b.upload_files("#upload-file-btn + input[type='file']", [files[-1]])
+
+            b.go(f'/files#/?path={dest_dir}')
+            b.wait_not_present('.pf-v5-c-empty-state')
+            b.wait_in_text(".pf-v5-c-alert__title", "Failed")
+            b.wait_in_text(".pf-v5-c-alert__description", "Not permitted to perform this action")
+            b.click(".pf-v5-c-alert__action button")
+            b.wait_not_present(".pf-v5-c-alert__action")
+            # temporary file does not exist
+            m.execute(f"! test -f /home/testuser/{os.path.basename(files[-1])}")
```

Comment:
This isn't the temp file, it's the real file name, no? Otherwise it'd have to be `.filename.tmp`. Either way, this check is a bit moot, as testuser/ is immutable.  (The "not permitted" above is the useful bit). But doesn't hurt to leave it in here, perhaps with something like `self.assertEqual(exe("ls /home/testuser"), "expected files")`? More robust than copying the temp file name schema here.